zetaclient/chains/evm/observer/observer.go

```diff
@@ -486,13 +475,13 @@ func (ob *Observer) LoadLastBlockScanned() error {
 
 // postBlockHeader posts the block header to zetacore
 // TODO(revamp): move to a block header file
-func (ob *Observer) postBlockHeader(tip uint64) error {
+func (ob *Observer) postBlockHeader(ctx context.Context, tip uint64) error {
 	bn := tip
 
-	res, err := ob.ZetacoreClient().GetBlockHeaderChainState(ob.Chain().ChainId)
-	if err == nil && res.ChainState != nil && res.ChainState.EarliestHeight > 0 {
+	chainState, err := ob.ZetacoreClient().GetBlockHeaderChainState(ctx, ob.Chain().ChainId)
+	if err == nil && chainState != nil && chainState.EarliestHeight > 0 {
```

Comment:
In the bitcoin `observer.go` doesn't check `chainState != nil`. For consistency purpose, can we simply use same criterion for both?